python/graphstorm/gpartition/metis_partition.py

```diff
@@ -0,0 +1,201 @@
+        ip_list: str
+            ip list
+        graph_name: str
+            Graph name
+        """
+        command = f"mpirun -np 1 --allow-run-as-root \
+                    --hostfile {ip_list} \
+                    --mca orte_base_help_aggregate 0 -mca btl_tcp_if_include eth0 \
+                    -wdir {input_path} \
+                ~/local/bin/pm_dglpart {graph_name} {num_parts} {input_path}/parmetis_nfiles.txt \
+                  {input_path}/parmetis_efiles.txt"
+
+        if self.run_command(command):
+            logging.info("Successfully execute parmetis process.")
+            return True
+        else:
+            logging.info("Failed to execute parmetis process.")
+            return False
+
+    def _launch_postprocess(self, num_parts, input_data_path, dgl_tool_path,
+                            metadata_filename, graph_name, partition_dir):
+        """ Launch postprocess which translates nid-partid mapping into
+            Per-node-type partid mappings.
+
+        Parameters
+        ----------
+        num_parts: int
+            Number of graph partitions
+        input_data_path: str
+            Path to the input graph data
+        dgl_tool_path: str
+            Path to the dgl tool added in the PYTHONPATH
+        metadata_filename: str
+            Meta data configuration name
+        graph_name: str
+            name of the graph in the parmetis step
+        partition_dir: str
+            output path
+        """
+        command = f"python3 {dgl_tool_path}/distpartitioning/parmetis_postprocess.py \
+                        --postproc_input_dir {input_data_path} \
+                        --schema_file {metadata_filename} \
+                        --parmetis_output_file {input_data_path}/{graph_name}_part.{num_parts} \
+                        --partitions_dir {partition_dir}"
+
+        if self.run_command(command):
+            logging.info("Successfully execute post parmetis process.")
+            return True
+        else:
+            logging.info("Failed to execute post parmetis process.")
+            return False
+
+    def run_command(self, command):
+        """Function to execute a command and check for its success."""
+        logging.info("Executing command: %s", command)
+        try:
+            # Execute the command and check if it completes successfully
+            result = subprocess.run(command, shell=True, check=True, text=True,
+                                    stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            logging.info("Command output: %s", result.stdout)
+            return True  # Return True if the command was successful
+        except subprocess.CalledProcessError as e:
+            logging.info("Error executing command: %s", e.stderr)
+            return False  # Return False if the command failed
+
+    def assigned_port(self, ip_file, port="2222"):
+        """Function to assigned port to each ip prepared for mpi."""
```

Comment:
mpi run needs that. It can only run mpi with specifically named port. Check here: https://github.com/open-mpi/ompi/issues/2224. But when we build ssh connection, the ip file with port specified is illegal. So we have to build two ip files. But the file with port specified is only available during parmetis, so I strongly suggest we keep the new ip file as an internal result only.

Reply:
It only runs on leader, once the leader creates the ip file, it should be able to connect the child worker node. The worker node here is only considered as a computing resource.